### PR TITLE
Added JWT token support for all internal service methods that require…

### PR DESCRIPTION
**Pull request by rtorres87**

### CODE REVIEW

File: src/main/java/com/plenigo/sdk/internal/services/InternalUserApiService.java [hunk 1]
```diff
@@ -107,11 +104,10 @@ public boolean isPaywallEnabled(String baseUrl, String secret, String companyId)
      */
     public UserData getUserData(String url, String companyId, String secret, String accessToken) throws PlenigoException {
         Map<String, Object> params = new HashMap<String, Object>();
-        params.put(ApiParams.COMPANY_ID, companyId);
-        params.put(ApiParams.SECRET, secret);
         params.put(ApiParams.ACCESS_TOKEN, accessToken);
         LOGGER.log(Level.FINEST, "Seeking user information for company: {0}", params.get(ApiParams.COMPANY_ID));
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/services/InternalUserApiService.java]: You should use comanyId cause now it isnt in params

File: src/main/java/com/plenigo/sdk/internal/util/Base64Util.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        try {
+            byte[] encode = DatatypeConverter.printBase64Binary(data).getBytes(Charset.DEFAULT);
+            for (int i = 0; i < encode.length; i++) {
+                if (encode[i] == '+') {
+                    encode[i] = '-';
+                } else if (encode[i] == '/') {
+                    encode[i] = '_';
+                }
+            }
+            return new String(encode);
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/Base64Util.java]: Why not new String(encode, Charset.DEFAULT);

File: src/main/java/com/plenigo/sdk/internal/util/HashUtils.java [hunk 3]
```diff
@@ -76,10 +76,10 @@ public static String calculateHash(final String data) throws PlenigoException {
     public static String calculateHMAC(String data, String secret) throws PlenigoException {
         try {
             Mac sha512HMAC = Mac.getInstance(HMAC_ALGORITHM);
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/HashUtils.java]: sha512HMAC?

File: src/main/java/com/plenigo/sdk/internal/util/JWT.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+import com.plenigo.sdk.PlenigoException;
+import org.json.simple.JSONValue;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+
+
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: Empty line

File: src/main/java/com/plenigo/sdk/internal/util/JWT.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+ * </p>
+ * <p>
+ * <strong>Thread safety:</strong> This class is thread safe and can be injected.
+ * </p>
+ */
+public final class JWT {
+    private static final int DEFAULT_EXPIRATION_TIME = (60 * 5);
+    public static final int MILLIS_IN_A_SECOND = 1000;
+    private static int expirationTime = DEFAULT_EXPIRATION_TIME;
+
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: Empty line

File: src/main/java/com/plenigo/sdk/internal/util/JWT.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+     * @throws PlenigoException if any exception happens during token creation
+     */
+    public static Map<String, String> generateJWTTokenHeader(String companyId, String secret) throws PlenigoException {
+        String jti = UUID.randomUUID().toString();
+        long exp = (System.currentTimeMillis() / MILLIS_IN_A_SECOND) + (expirationTime); //add 5minutes to current time
+        String token = "";
+        if (companyId != null && !companyId.isEmpty() && secret != null && !secret.isEmpty()) {
+            token = generateJWTToken(companyId, secret, jti, exp);
+        }
+        return Collections.singletonMap("plenigoToken", token);
```
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: Why adding an empty token? (when comanyId is null)

File: src/test/java/com/plenigo/sdk/internal/util/JWTTest.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package com.plenigo.sdk.internal.util;
+
+import com.plenigo.sdk.PlenigoException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+
```
Comment (robertholfeld) [inline src/test/java/com/plenigo/sdk/internal/util/JWTTest.java]: Empty line

File: src/main/java/com/plenigo/sdk/internal/services/InternalUserApiService.java (hunk 1 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/services/InternalUserApiService.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/util/Base64Util.java (hunk 2 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/Base64Util.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/util/HashUtils.java (hunk 3 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/HashUtils.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/util/JWT.java (hunk 4 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/util/JWT.java (hunk 5 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/util/JWT.java (hunk 6 above)
Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: Just to return something, there is no other default value that i could think of

File: src/test/java/com/plenigo/sdk/internal/util/JWTTest.java (hunk 7 above)
Comment (rtorres87) [inline src/test/java/com/plenigo/sdk/internal/util/JWTTest.java]: Fixed

File: src/main/java/com/plenigo/sdk/internal/util/JWT.java (hunk 6 above)
Comment (robertholfeld) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: But you can return an empty map, cause if the map is not empty you send it in the header and without a token the plenigoToken is useless

Comment (rtorres87) [inline src/main/java/com/plenigo/sdk/internal/util/JWT.java]: aah okay, ill return an empty map then

**Pull request merged**